mpags-cipher.cpp

```diff
@@ -0,0 +1,142 @@
+#include <iostream>
+#include <vector>
+#include <string>
+
+int main(int argc, char* argv[]){
+
+    // Convert command line arguments into an easier to use form
+    const std::vector<std::string> cmdLineArgs{argv, argv+argc};
+    const std::size_t nCmdLineArgs(cmdLineArgs.size());
+
+    // Help requested boolean and message
+    bool helpRequested{false};
+    std::string help_message{
+        """Usage: ./mpags-cipher [options]\n"""
+        """Options:\n"""
+        """ -h, --help: Display this message\n"""
+        """ --version: Display version\n"""
+        """ -i FILE  ,  --input-file FILE: Specify the input file\n"""
+        """ -o FILE  ,  --output-file FILE: Specify the output file"""
+        };
+
+    // Versioning
+    bool versionRequested{false};
+    const std::string version_number{"0.1.0"};
+
+    // Input file
+    std::string input_file{""};
+    bool input_file_given{false};
+
+    // Output file
+    std::string output_file{""};
+    bool output_file_given{false};
+
+    for (size_t i{1}; i < nCmdLineArgs; i++){
+        if (cmdLineArgs[i] == "--help" || cmdLineArgs[i] == "-h"){
+            helpRequested = true;
+        }
+        else if (cmdLineArgs[i] == "--version"){
+            versionRequested = true;
+        }
+        else if (cmdLineArgs[i] == "--input-file" || cmdLineArgs[i] == "-i"){
+            // Check if the list is long enough
+            if (i < nCmdLineArgs - 1){
+                input_file = cmdLineArgs[i+1];
+                input_file_given = true;
+                i++; // Increment so that the argument isn't reparsed
+            }
+            else{
+                std::cout << "Error: No input file specified" << std::endl;
+                return 1;
+            }
+        }
+        else if (cmdLineArgs[i] == "--output-file" || cmdLineArgs[i] == "-o"){
+            // Check if the list is long enough
+            if (i < nCmdLineArgs - 1){
+                output_file = cmdLineArgs[i+1];
+                output_file_given = true;
+                i++; // Increment so that the argument isn't reparsed
+            }
+            else{
+                std::cout << "Error: No output file specified" << std::endl;
+                return 1;
+            }
+        }
+        else{
+            std::cout << "Argument " << cmdLineArgs[i] << " is unknown. Use ./mpags-cipher --help to list configurations" << std::endl;
```

Comment:
Great that you're printing an error if an unknown argument is supplied.  Just one small suggestion - instead of having `./mpags-cipher` hardcoded into the string, it would be better to use `cmdLineArgs[0]`, since you don't know exactly how the program is being executed.